pythran/optimizations/pattern_transform.py

```diff
@@ -125,3 +135,21 @@ def visit(self, node):
                 node = PlaceholderReplace(check.placeholders).visit(replace())
                 self.update = True
         return super(PatternTransform, self).visit(node)
+
+    def visit_BinOp(self, node):
+        """ 
+        Special method for BinOp.
+        Try to replace if node match the given pattern. 
+        """
```

Comment:
why do you need that specialization?

Reply:
I thought would be good for speeding-up the optimization.
As soon as I find a way to implement safely the patterns I have in mind, we may have order of  100 patterns to check for replacement.  
Then, as it was,  pattern transform would check for any node any given pattern careless of the structure.
In this way, instead, it goes trough BinOp just when needed and select from the dictionary the patterns relative to the used operator.